patroni/config.py

```diff
@@ -393,12 +393,12 @@ def _set_section_values(section, params):
                 if value is not None:
                     ret[first][second] = value
 
-        for second in ('max_queue_size', 'file_size', 'file_num'):
-            value = ret.get('log', {}).pop(second, None)
+        for first, second in (('rest_api', ('request_queue_size',)), ('log', ('max_queue_size', 'file_size', 'file_num'))):
```

Comment:
```suggestion
        for first, params in (('restapi', ('request_queue_size',)),                                                                    
                              ('log', ('max_queue_size', 'file_size', 'file_num'))):                                                   
```

And you need to have a second loop over params:
```python
for second in params:
    ...
```

Reply:
Added request_queue_size to _set_section_values() and fixed restapi related loop in _build_environment_configuration(). Hope it looks better now.